src/components/NoteDetails/NoteDetailsSagas.js

```diff
@@ -145,29 +147,57 @@ function* applyQueuedNoteUpdate({ data: { note, queuedChanges: changes } }) {
 }
 
 function* requestNoteNameUpdate({ data: { note } }) {
-  const newNote = yield noteStorage.updateNoteName({
-    note,
-    newName: note.name,
-  });
-  yield put(noteNameUpdateRequestSuccessAction(newNote));
-  queuedChanges.noteName = null;
+  try {
+    const newNote = yield noteStorage.updateNoteName({
+      note,
+      newName: note.name,
+    });
+    yield put(noteNameUpdateRequestSuccessAction(newNote));
+    queuedChanges.noteName = null;
+  } catch (e) {
+    yield put(
+      noteNameUpdateRequestFailureAction({
+        name: e.name,
+        message: e.message,
+        text: 'Failed to save changes note name',
+      }),
+    );
+    console.error(e);
+  }
 }
 
 function* requestNoteContentUpdate({ data: { note } }) {
   try {
-    const newNote = yield noteStorage.updateNoteContent(note);
-    yield put(noteContentUpdateRequestSuccessAction(newNote));
+    yield noteStorage.updateNoteContent(note); // func return undefined or error
+    yield put(noteContentUpdateRequestSuccessAction());
     queuedChanges.noteContent = null;
-  } catch (error) {
-    toast.error('Failed to save note content');
-    console.error(error);
+  } catch (e) {
+    yield put(
+      noteContentUpdateRequestFailureAction({
+        name: e.name,
+        message: e.message,
+        text: 'Failed to save changes note content',
+      }),
+    );
+    console.error(e);
   }
 }
 
 function* createNoteRequest({ data: { note } }) {
-  const newNote = yield noteStorage.create(note);
-  newNote.parent = note.parent;
-  yield put(createNoteSuccessAction(newNote));
+  try {
+    const newNote = yield noteStorage.create(note); // func return object (not Promise) or error
```

Comment:
If it doesn't return promise, why are we using `yield`? 
I believe it does return a promise, because it's an asynchronous operation.